src/core/test/datagen/src/main/scala/GenerateDataset.scala

```diff
@@ -1,110 +0,0 @@
-// Copyright (C) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See LICENSE in project root for information.
-
-package com.microsoft.ml.spark
-
-import com.microsoft.ml.spark.ColumnOptions.ColumnOptions
-import com.microsoft.ml.spark.DataOptions.DataOptions
-import org.apache.spark.mllib.random.RandomRDDs
-import org.apache.spark.sql._
-import org.apache.spark.sql.types._
-
-import scala.util.Random
-
-/** Defines methods to generate a random spark DataFrame dataset based on given options. */
-object GenerateDataset {
-
-  /** Generates a random Spark DataFrame given a set of dataset generation constraints.
-    * @param sparkSession The spark session.
-    * @param datasetGenerationConstraints The dataset generation constraints to use.
-    * @param seed The random seed.
-    * @return A randomly generated dataset.
-    */
-  def generateDataset(sparkSession: SparkSession,
```

Comment:
Sorry for removing a lot of your code, right now its temporary as it needs to be re-worked. That fuzzing API was not general enough and required a lot of information at class extension time. All of the random test generation stuff can be easily added in the new API as a class that extends the ExperimentFuzzing trait. More specifically, the new API takes a sequence of stages, fittingDFs, and trainingDFs. These can be user supplied or in your case, generated with the generation options. This way its easy to either supply a simple test on one concrete dataframe, or in your case a large suite of random tests.

In the early stages i think we should focus on a simple and extensible API and then build the complex dataset generation stuff on top of it. We need a good way to quickly test the basic functionality with minimal overhead in the test class. Once we scale up our quality measures we can consider more rigorous schema based fuzzing.